src/python/pants/engine/internals/graph.py

```diff
@@ -325,10 +319,38 @@ async def _parametrized_target_generators_with_templates(
             f"so target generator {address} (with type {target_type.alias}) cannot "
             f"parametrize the {generator_fields_parametrized_text} {noun}."
         )
+    return template_fields, generator_fields
+
+
+async def _parametrized_target_generators_with_templates(
+    address: Address,
+    target_adaptor: TargetAdaptor,
+    target_type: type[TargetGenerator],
+    template_fields_from_generator: Mapping[str, Any],
+    generator_fields: dict[str, Any],
+    union_membership: UnionMembership,
+) -> list[tuple[TargetGenerator, dict[str, Any]]]:
+    # Pre-load field values from defaults for the target type being generated.
+    if hasattr(target_type, "generated_target_cls"):
+        family = await Get(AddressFamily, AddressFamilyDir(address.spec_path))
+        generated_target_classes = (
+            target_type.generated_target_cls
+            if isinstance(target_type.generated_target_cls, tuple)
+            else (target_type.generated_target_cls,)
```

Comment:
I'd prefer this be encapsulated in a class property (or class method), as it's a code smell for me that `generated_target_cls` could be plural.

Trying to be as backwards compatible as possible, what do you think about having a new class var, say `generated_target_classes: ClassVar[tuple[type[Target], ...]]` and then define a class property for `generated_target_cls` which just returns the first type, and another class property, `generated_target_types` which returns `generated_target_classes` if defined, otherwise `(cls.generated_target_cls,)`. This way, we shouldn't break anyone, you can define either `generated_target_cls` or `generated_target_classes` in your subclasses.

(not sure if mypy will be happy with this setup, though.. as it's a bit on the tricky side)

Reply:
If we do this should we break them out into separate classes i.e. 

`TargetGeneratorBase` -> has abstract class method/property to get generated target classes tuple
`TargetGenerator` -> has `generated_target_cls` and implements the above as `(generated_target_cls,)`
`MultiTypeTargetGenerator` -> has `generated_target_classes` and just returns this in the above